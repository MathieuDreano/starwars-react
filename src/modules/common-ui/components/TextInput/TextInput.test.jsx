import React from 'react';
import TestRenderer from 'react-test-renderer';

import { TextInput } from '.';

it('renders correctly a TextInput component', () => {
  const testRenderer = TestRenderer.create(<TextInput value="Hello" />);
  expect(testRenderer.toJSON()).toMatchSnapshot();
});

it('renders correctly a input component with onChange', () => {
  const mockOnChange = jest.fn();
  const testRenderer = TestRenderer.create(<TextInput onChange={mockOnChange} value="Hello" />);
  const testInstance = testRenderer.root;
  const inputIntance = testInstance.findByType('input');
  const inputProps = inputIntance.props;
  expect(inputProps.className).toBe('input');

  inputProps.onChange();
  expect(mockOnChange).toHaveBeenCalled();
});

it('renders correctly a input component with custom className', () => {
  const testRenderer = TestRenderer.create(<TextInput className="salut" value="Hello" />);
  const testInstance = testRenderer.root;
  const inputInstance = testInstance.findByType('input');
  const inputProps = inputInstance.props;
  expect(inputProps.className).toBe('input salut');
});
