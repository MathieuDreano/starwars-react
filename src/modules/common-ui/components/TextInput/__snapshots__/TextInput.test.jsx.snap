// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly a TextInput component 1`] = `
<input
  className="input"
  type="text"
  value="Hello"
/>
`;
